be1-go/hub/socket_types.go

```diff
@@ -0,0 +1,56 @@
+package hub
+
+import (
+	"github.com/gorilla/websocket"
+	"sync"
+)
+
+// socket types
+type SocketType string
+
+const (
+	clientSocket SocketType = "client"
+	organizerSocket SocketType= "organizer"
+	witnessSocket SocketType= "witness"
+)
+
+func newSocket(socketType SocketType, h Hub, conn *websocket.Conn) *baseSocket {
+	return &baseSocket {
```

Comment:
Where is the struct `baseSocket` defined? I would've expected to find it in this file, am I missing smth?

Reply:
I defined it in socket.go, since all of the methods of Socket (interface) are implemented on baseSocket. I thought the socket_types should just contain the code relevant to different socket types. Do you think I should move baseSocket to socketType ?